File: plugins/codemodder-plugin-maven/src/main/java/io/codemodder/plugins/maven/operator/QueryByInvoker.java
package io.codemodder.plugins.maven.operator;

import java.io.File;
import org.apache.maven.shared.invoker.DefaultInvoker;
import org.apache.maven.shared.invoker.InvocationRequest;
import org.apache.maven.shared.invoker.InvocationResult;
import org.apache.maven.shared.invoker.MavenInvocationException;

class QueryByInvoker extends AbstractQueryCommand {

  @Override
  protected void extractDependencyTree(File outputPath, File pomFilePath, ProjectModel c) {
    DefaultInvoker invoker = new DefaultInvoker();

    InvocationRequest invocationRequest = buildInvocationRequest(outputPath, pomFilePath, c);

    InvocationResult invocationResult;
    try {
      invocationResult = invoker.execute(invocationRequest);
    } catch (MavenInvocationException e) {
      throw new RuntimeException(e);
    }

    int exitCode = invocationResult.getExitCode();

    if (exitCode != 0) {
      throw new IllegalStateException("Unexpected Status Code from Invoker: " + exitCode);
    }
  }
}

File: src/main/java/io/github/pixee/maven/operator/QueryByInvoker.kt
package io.github.pixee.maven.operator

import org.apache.maven.shared.invoker.DefaultInvoker
import org.apache.maven.shared.invoker.InvocationRequest
import java.io.File

class QueryByInvoker : AbstractQueryCommand() {

    override fun extractDependencyTree(
        outputPath: File,
        pomFilePath: File,
        c: ProjectModel
    ) {
        val invoker = DefaultInvoker()

        val invocationRequest: InvocationRequest =
            buildInvocationRequest(outputPath, pomFilePath, c)

        val invocationResult = invoker.execute(invocationRequest)

        val exitCode = invocationResult.exitCode

        if (0 != exitCode) {
            throw IllegalStateException(
                "Unexpected Status Code from Invoker: %02d".format(
                    exitCode
                )
            )
        }
    }

}